pkg/ai/iai.go

```diff
@@ -30,12 +30,15 @@ type IAIConfig interface {
 	GetPassword() string
 	GetModel() string
 	GetBaseURL() string
+	GetEngine() string
 }
 
```

Comment:
I would suggest doing something like 
```
var (
backends := []IAI { 
 &OpenAIClient{}, 
&AzureAIClient{}, 
...
}
```

Then in the new client, create a loop 

```
for b, := backend {
 // if it matches b.GetName() return
}

````

Reply:
I updated the PR, based on your feedback